pkgs/development/interpreters/ruby/parse_version.nix

```diff
@@ -0,0 +1,91 @@
+# Contains the ruby version heuristics
+{ lib }:
+with lib;
+let
+  parseVersion = versionStr:
+    let
+      pair = splitString "-" versionStr;
```

Comment:
From the stdlib (regarding splitString): "NOTE: this function is not performant and should never be used." :'(